hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSEditLogOp.java

```diff
@@ -2946,7 +2946,7 @@ void fromXml(Stanza st) throws InvalidXmlException {
       this.newLength = Long.parseLong(st.getValue("NEWLENGTH"));
       this.timestamp = Long.parseLong(st.getValue("TIMESTAMP"));
       if (st.hasChildren("BLOCK"))
-        this.truncateBlock = FSEditLogOp.blockFromXml(st);
+        this.truncateBlock = FSEditLogOp.blockFromXml(st.getChildren("BLOCK").get(0));
```

Comment:
Good catch here. I think it is better to fix it at org.apache.hadoop.hdfs.server.namenode.FSEditLogOp#blockFromXml. Thanks.

Reply:
After another review, I think it is proper now. Do you mind to add new unit test to cover this case?

Reply:
After another review, I think it is proper now. Do you mind to add new unit test to cover this case?

Reply:
thanks @Hexiaoqiao  for your review, will add test later